File: SiteVisitsNetCore2020Web/ClientApp/src/app/models/Device.ts
export class Device {
  //id: number;
  id: string;
  operatingSystem: string;
  resolution: string;

  constructor(options: {
    //id: number,
    id: string,
    operatingSystem: string,
    resolution: string
  }) {
    if (options) {
      this.id = options.id;
      this.operatingSystem = options.operatingSystem;
      this.resolution = options.resolution;
    }
  }

}

File: SiteVisitsNetCore2020Web/Models/Device.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace SiteVisitsNetCore2020Web.Models
{
    public class Device
    {
        //public int Id { get; set; }
        public Guid Id { get; set; }
        public string OperatingSystem { get; set; }
        public string Resolution { get; set; }
        public DateTime FirstOccurrence { get; set; }
        public DateTime DateCreated { get; set; }
    }
}
